src/routes/learn/$slug/_layout.tsx

```diff
@@ -24,24 +30,35 @@ const getModulesWithSegmentsFn = createServerFn()
   });
 
 export const modulesQueryOptions = queryOptions({
-  queryKey: ["modules"],
+  queryKey: ['modules'],
   queryFn: () => getModulesWithSegmentsFn(),
 });
 
-export const getProgressFn = createServerFn()
+export const getProgressForUserFn = createServerFn()
   .middleware([unauthenticatedMiddleware])
   .handler(async ({ context }) => {
     return context.userId ? getAllProgressForUserUseCase(context.userId) : [];
   });
 
-export const Route = createFileRoute("/learn/$slug/_layout")({
+export const Route = createFileRoute('/learn/$slug/_layout')({
   component: RouteComponent,
   loader: async ({ context: { queryClient }, params }) => {
     const { segment } = await getSegmentInfoFn({ data: { slug: params.slug } });
     const isPremium = await isUserPremiumFn();
     const isAdmin = await isAdminFn();
-    const progress = await getProgressFn();
-    await queryClient.ensureQueryData(modulesQueryOptions);
+    const progress = await getProgressForUserFn();
+    // const segmentProgress = await getSegmentProgressFn({
+    //   data: { segmentId: segment.id },
+    // });
+    // console.log(segmentProgress);
+    queryClient.ensureQueryData(modulesQueryOptions);
+    queryClient.ensureQueryData({
+      queryKey: ['progress', segment.id] as const,
+      queryFn: ({ queryKey }) =>
+        getCompletedUserProgressForSegmentFn({
+          data: { segmentId: queryKey[1] },
+        }),
+    });
```

Comment:
Same as in video-header file
using `queryOptions` to extract this into it's own function to reduce code duplication
https://github.com/webdevcody/tanstack-course-platform/blob/7e591ae9fc824f8add7d13720a24f56e79e9f974/src/lib/queries/comments.ts#L4-L9